molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/download/DataExplorerDownloadHandler.java

```diff
@@ -0,0 +1,112 @@
+package org.molgenis.dataexplorer.download;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.DataService;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.Repository;
+import org.molgenis.data.Writable;
+import org.molgenis.data.WritableFactory;
+import org.molgenis.data.csv.CsvWriter;
+import org.molgenis.data.excel.ExcelWriter;
+import org.molgenis.data.excel.ExcelWriter.FileFormat;
+import org.molgenis.data.support.QueryImpl;
+import org.molgenis.dataexplorer.controller.DataRequest;
+import org.molgenis.dataexplorer.controller.DataRequest.ColNames;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+
+public class DataExplorerDownloadHandler
+{
+
+	private DataService dataService;
+
+	@Autowired
+	public DataExplorerDownloadHandler(DataService dataService)
+	{
+		this.dataService = dataService;
+	}
+
+	public void writeToExcel(DataRequest dataRequest, OutputStream outputStream, char separator) throws IOException
+	{
+		WritableFactory writableFactory = new ExcelWriter(outputStream, FileFormat.XLSX);
+
+		Repository repo = dataService.getRepository(dataRequest.getEntityName());
+
+		String entityName = dataRequest.getEntityName();
+
+		Iterable<AttributeMetaData> attributes = dataService.getEntityMetaData(entityName).getAtomicAttributes();
+		List<String> attributeNames = new ArrayList<String>();
+
+		attributes.forEach(attribute -> attributeNames.add(attribute.getName()));
+
+		try
+		{
+			Writable writable = writableFactory.createWritable(entityName, attributeNames);
+			writable.add(repo);
```

Comment:
The CSV exporter filters using query, the Excel exporter writes the entire repo.